fhir-search/src/main/java/com/ibm/fhir/search/sort/Sort.java

```diff
@@ -81,72 +85,77 @@ public static Direction fromValue(String value) {
-                SearchConstants.Type sortParmType =
-                        SearchConstants.Type.fromValue(sortParmProxy.getType().getValue());
-                SortParameter sortParm = new SortParameter(sortParmCode, sortParmType, sortDirection);
+                // Per the FHIR spec, the _sort parameter value is a search parameter. We need to determine what
+                // type of search parameter.
+                SearchParameter sortParmProxy = SearchUtil.getSearchParameter(resourceTypeName, sortParmCode);
+                if (!isUndefinedOrLenient(resourceTypeName, sortParmCode, sortParmProxy, context)) {
+                    SearchConstants.Type sortParmType =
+                            SearchConstants.Type.fromValue(sortParmProxy.getType().getValue());
+                    SortParameter sortParm = new SortParameter(sortParmCode, sortParmType, sortDirection);
 
-                checkSystemLevel(resourceTypeName, sortParm.getCode());
+                    checkSystemLevel(resourceTypeName, sortParm.getCode(), context);
 
-                context.getSortParameters().add(sortParm);
+                    context.getSortParameters().add(sortParm);
+                }
+            } catch (FHIRSearchException se) {
+                // If we're in lenient mode and there was an issue parsing the _sort parameter value then log and move on to the next one.
+                if (context.isLenient()) {
+                    String msg = "Sort value '" + sortParmCode + "' for resource type '" + resourceTypeName + "' ignored";
+                    log.log(Level.FINE, msg, se);
+                    context.addOutcomeIssue(FHIRUtil.buildOperationOutcomeIssue(IssueSeverity.WARNING, IssueType.INCOMPLETE, msg));
+                } else {
+                    throw se;
+                }
             }
         }
     }
 
     /**
      * checks to see if undefined as a SearchParameter or Lenient
      *
-     * @param resourceTypeName
-     * @param sortParmCode
-     * @param sortParmProxy
-     * @param lenient
-     * @return
+     * @param resourceTypeName the resource type
+     * @param sortParmCode the sort value
+     * @param sortParmProxy the search parameter
+     * @param context the context
+     * @return true or false
      * @throws FHIRSearchException
      */
     public boolean isUndefinedOrLenient(String resourceTypeName, String sortParmCode, SearchParameter sortParmProxy,
-            boolean lenient) throws FHIRSearchException {
+            FHIRSearchContext context) throws FHIRSearchException {
         boolean result = sortParmProxy == null;
         if (result) {
             String msg = buildUndefinedSortParamMessage(resourceTypeName, sortParmCode);
-
-            // Stop Processing if not lenient
-            if (!lenient) {
-                throw SearchExceptionUtil.buildNewInvalidSearchException(msg);
+            if (context.isLenient()) {
+                log.fine(msg);
```

Comment:
isLoggable?

Reply:
The log.fine(...) method internally does an isLoggable check.